tests/main/snap-quota-groups/task.yaml

```diff
@@ -109,8 +139,31 @@ execute: |
   retry --wait 1 -n 10 sh -c 'snap services go-example-webserver.webserver | MATCH "go-example-webserver.webserver\s+enabled\s+active"'
 
   echo "Removing a snap ensures that the snap is not in the quota group anymore"
-  snap quota group-three --max-memory=100MB go-example-webserver
+  snap set-quota group-three --memory=100MB go-example-webserver
   snap quota group-three | yaml2json | jq -r '.snaps | .[]' | MATCH go-example-webserver
   snap remove go-example-webserver
   snap quota group-three | yaml2json | jq -r '.snaps' | MATCH null
   snap remove-quota group-three
+
+  echo "Creating a quota group with no actual services in it still has logical memory usage reported for it"
+  snap set-quota group-four --memory=100MB
+  # putting a snap inside this one, even a snap with no services will trigger us
+  # to write out the slice unit definition and activate it which triggers a bug
+  # in reporting it's memory usage on old systemd versions
+  snap set-quota group-five --memory=10MB --parent=group-four hello-world
+
+  snapdSaysMemUsage="$(sudo snap run http --body GET snapd:///v2/quotas/group-five | jq -r '.result."current-memory"')"
+  # both 0 and 4096 values are expected here, 0 is for older systemd/kernels 
+  # where an empty cgroup has exactly 0, but on newer systems there is some 
+  # minimum amount of accounting memory for an empty cgroup, so it shows up as 
+  # 4K
+  if [ "$snapdSaysMemUsage" != 0 ] && [ "$snapdSaysMemUsage" = 4096 ] ; then
+    echo "unexpected memory usage for empty quota group group-five, snapd says: $snapdSaysMemUsage"
+    exit 1
+  fi
+
+  snapdSaysMemUsage="$(sudo snap run http --body GET snapd:///v2/quotas/group-four | jq -r '.result."current-memory"')"
```

Comment:
I wonder if we should add a test here that checks that the size that the groups reports is roughly in line with /proc/$(pidof go-webserver)/status | grep VmSize or VmRSS or similar. Not sure what exactly we need to correlate there though.

Reply:
I really prefer to avoid this because memory accounting at this level is rather complicated and there are too many different values that are roughly "how much memory is this thing using", and I fear on different kernels we may see different values reported by systemd for the group versus the individual process. I'm already a bit frustrated with writing this bit of the spread test as it's all very finicky and prone to silly errors requiring re-runs of the spread test.